ecom_test/src/generic_helpers/generic_coupon_helper.py

```diff
@@ -0,0 +1,57 @@
+
+
+import random
+from datetime import datetime
+import string
+from ecom_test.src.api_helpers.CouponsAPIHelper import CouponsAPIHelper
+
+
+class GenericCouponHelper:
+
+    def __init__(self):
+        self.coupon_api_helper = CouponsAPIHelper()
+
+    def create_coupon(self, coupon_code=None, length=4, expired=False, discount_type="percent", amount="100"):
+
+        if expired:
+            expiration_date = datetime.now().isoformat()
+        else:
+            expiration_date = None
+
+        if not coupon_code:
+            coupon_code = ''.join(random.choice(string.ascii_uppercase) for i in range(length))
+
+        payload = {
+            "code": coupon_code,
+            "discount_type": "percent",
```

Comment:
You have a parameter for 'discount_type' so you should not hard code "percent" here. use the parameter.
Same for 'amount'.

Reply:
Yes, you are right. I changed